src/v2.rs

```diff
@@ -176,24 +182,45 @@ impl Party {
         key_ids: &[u32],
         nonces: &[PublicNonce],
     ) -> SignatureShare {
-        self.sign_with_tweak(msg, party_ids, key_ids, nonces, &Scalar::from(0))
+        self.sign_with_tweak(msg, party_ids, key_ids, nonces, None)
     }
 
-    /// Sign `msg` with this party's shares of the group private key, using the set of `party_ids`, `key_ids` and corresponding `nonces` with a tweaked public key
+    /// Sign `msg` with this party's shares of the group private key, using the set of `party_ids`, `key_ids` and corresponding `nonces` with a tweaked public key. The posible values for tweak are
+    /// None    - standard FROST signature
+    /// Some(0) - BIP-340 schnorr signature using 32-byte private key adjustments
+    /// Some(t) - BIP-340 schnorr signature with BIP-341 tweaked keys, using 32-byte private key adjustments
     #[allow(non_snake_case)]
     pub fn sign_with_tweak(
         &self,
         msg: &[u8],
         party_ids: &[u32],
         key_ids: &[u32],
         nonces: &[PublicNonce],
-        tweak: &Scalar,
+        tweak: Option<Scalar>,
     ) -> SignatureShare {
-        let tweaked_public_key = self.group_key + tweak * G;
+        // When using BIP-340 32-byte public keys, we have to invert the private key if the public key is odd.  But if we're also using BIP-341 tweaked keys, we have to do the same thing if the tweaked public key is odd.  In that case, only invert the public key if exactly one of the internal or tweaked public keys is odd
```

Comment:
You should probably break this comment up into multiple lines, it looks like its at least 200 characters long. I use the rewrap VS Code extension to automatically do that for me (https://marketplace.visualstudio.com/items?itemName=stkb.rewrap, with the `opt+q` save me a ton of time).